### PR TITLE
Fix NaN values into inflow

### LINKED ISSUE #241: NaN values into inflows
## Description

While using pypsa-africa, for some plants in Nigeria, the inflow was completely nan, see [issue](https://github.com/pypsa-meets-africa/pypsa-africa/issues/108#issuecomment-1146717666)
In particular, when considering [this hydrodam](https://goo.gl/maps/xb6bdNnDoA3ZmhMi7) and the level 08 of the [Africa hydrobasins](https://www.hydrosheds.org/page/hydrobasins), the output of the inflow was completely NaN values.

image

## Expected Behavior

A non-nan vector

## Actual Behavior

A list of NaN values

## Your Environment

* The `atlite` version used:
* How you installed `atlite` (`conda`, `pip` or `github`):
* Operating System:
* My environment:
    <details>
      <summary>(output of `conda list`)</summary>
      ```
      _libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
affine                    2.3.1              pyhd8ed1ab_0    conda-forge
alsa-lib                  1.2.3.2              h166bdaf_0    conda-forge
amply                     0.1.5              pyhd8ed1ab_0    conda-forge
anyio                     3.6.1           py310hff52083_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py310h5764c6d_2    conda-forge
asttokens                 2.0.5              pyhd8ed1ab_0    conda-forge
atlite                    0.2.7              pyhd8ed1ab_0    conda-forge
attr                      2.5.1                h166bdaf_0    conda-forge
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
babel                     2.10.1             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
bleach                    5.0.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.1               h83bc5f7_3    conda-forge
bokeh                     2.4.3           py310hff52083_0    conda-forge
boost-cpp                 1.74.0               h6cacc03_7    conda-forge
bottleneck                1.3.4           py310hde88566_1    conda-forge
branca                    0.5.0              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h166bdaf_7    conda-forge
brotli-bin                1.0.9                h166bdaf_7    conda-forge
brotlipy                  0.7.0           py310h5764c6d_1004    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2022.5.18.1          ha878542_0    conda-forge
cairo                     1.16.0            ha12eb4b_1010    conda-forge
cartopy                   0.20.2          py310hb89acbc_4    conda-forge
cdsapi                    0.5.1              pyhd8ed1ab_0    conda-forge
certifi                   2022.5.18.1     py310hff52083_0    conda-forge
cffi                      1.15.0          py310h0fdd8cc_0    conda-forge
cfgv                      3.3.1              pyhd8ed1ab_0    conda-forge
cfitsio                   4.1.0                hd9d235c_0    conda-forge
cftime                    1.6.0           py310hde88566_1    conda-forge
charset-normalizer        2.0.12             pyhd8ed1ab_0    conda-forge
click                     8.1.3           py310hff52083_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.1.0              pyhd8ed1ab_0    conda-forge
coin-or-cbc               2.10.8               h3786ebc_0    conda-forge
coin-or-cgl               0.60.6               he2f9439_0    conda-forge
coin-or-clp               1.17.6               h256e9bb_3    conda-forge
coin-or-osi               0.108.7              h3b589db_0    conda-forge
coin-or-utils             2.11.6               h573740c_0    conda-forge
coincbc                   2.10.8            0_metapackage    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
colorcet                  3.0.0              pyhd8ed1ab_0    conda-forge
conda                     4.12.0          py310hff52083_0    conda-forge
conda-package-handling    1.8.1           py310h5764c6d_1    conda-forge
configargparse            1.5.3              pyhd8ed1ab_0    conda-forge
connection_pool           0.0.3              pyhd3deb0d_0    conda-forge
country_converter         0.7.4              pyhd8ed1ab_0    conda-forge
countrycode               0.2                      pypi_0    pypi
cryptography              37.0.2          py310h597c629_0    conda-forge
curl                      7.83.1               h7bff187_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.11.2          py310h5764c6d_2    conda-forge
dask                      2022.5.2           pyhd8ed1ab_0    conda-forge
dask-core                 2022.5.2           pyhd8ed1ab_0    conda-forge
datashader                0.14.0             pyh6c4a22f_0    conda-forge
datashape                 0.5.4                      py_1    conda-forge
datrie                    0.8.2           py310h6acc77f_3    conda-forge
dbus                      1.13.6               h5008d03_3    conda-forge
debugpy                   1.6.0           py310hd8f1fbe_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
deprecation               2.1.0              pyh9f0ad1d_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
distlib                   0.3.4              pyhd8ed1ab_0    conda-forge
distributed               2022.5.2           pyhd8ed1ab_0    conda-forge
docutils                  0.18.1          py310hff52083_1    conda-forge
dpath                     2.0.6           py310hff52083_1    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
entsoe-py                 0.5.4                    pypi_0    pypi
esy-osm-pbf               0.0.1                    pypi_0    pypi
esy-osmfilter             1.0.11                   pypi_0    pypi
et_xmlfile                1.0.1                   py_1001    conda-forge
executing                 0.8.3              pyhd8ed1ab_0    conda-forge
expat                     2.4.8                h27087fc_0    conda-forge
fftw                      3.3.10          nompi_h77c792f_102    conda-forge
filelock                  3.7.1              pyhd8ed1ab_0    conda-forge
fiona                     1.8.20          py310hfb73927_4    conda-forge
flit-core                 3.7.1              pyhd8ed1ab_0    conda-forge
folium                    0.12.1.post1       pyhd8ed1ab_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.14.0               h8e229c2_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.33.3          py310h5764c6d_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
freexl                    1.0.6                h7f98852_0    conda-forge
fsspec                    2022.5.0           pyhd8ed1ab_0    conda-forge
future                    0.18.2          py310hff52083_5    conda-forge
gdal                      3.4.3                    pypi_0    pypi
geographiclib             1.52               pyhd8ed1ab_0    conda-forge
geopandas                 0.10.2             pyhd8ed1ab_1    conda-forge
geopandas-base            0.10.2             pyha770c72_1    conda-forge
geopy                     2.2.0              pyhd8ed1ab_0    conda-forge
geos                      3.10.2               h9c3ff4c_0    conda-forge
geotiff                   1.7.1                h509b78c_1    conda-forge
geoviews                  1.9.5              pyhd8ed1ab_0    conda-forge
geoviews-core             1.9.5              pyha770c72_0    conda-forge
gettext                   0.19.8.1          h73d1719_1008    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
gitdb                     4.0.9              pyhd8ed1ab_0    conda-forge
gitpython                 3.1.27             pyhd8ed1ab_0    conda-forge
glpk                      5.0                  h445213a_0    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
googledrivedownloader     0.4                pyhd3deb0d_1    conda-forge
gst-plugins-base          1.18.5               hf529b03_3    conda-forge
gstreamer                 1.20.2               hd4edc92_1    conda-forge
gurobi                    9.5.1                   py310_0    gurobi
hdf4                      4.2.15               h10796ff_3    conda-forge
hdf5                      1.12.1          nompi_h2386368_104    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
holoviews                 1.14.9             pyhd8ed1ab_0    conda-forge
hvplot                    0.8.0              pyh6c4a22f_0    conda-forge
icu                       69.1                 h9c3ff4c_0    conda-forge
identify                  2.5.1              pyhd8ed1ab_0    conda-forge
idna                      3.3                pyhd8ed1ab_0    conda-forge
importlib-metadata        4.11.4          py310hff52083_0    conda-forge
importlib_metadata        4.11.4               hd8ed1ab_0    conda-forge
importlib_resources       5.7.1              pyhd8ed1ab_1    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
ipopt                     3.13.2               h7839693_1    conda-forge
ipykernel                 6.13.0          py310hfdc917e_0    conda-forge
ipython                   8.4.0           py310hff52083_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jack                      1.9.18            hfd4fe87_1001    conda-forge
jedi                      0.18.1          py310hff52083_1    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h166bdaf_1    conda-forge
json-c                    0.16                 hc379101_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                4.6.0              pyhd8ed1ab_0    conda-forge
jupyter_client            7.3.1              pyhd8ed1ab_0    conda-forge
jupyter_core              4.10.0          py310hff52083_0    conda-forge
jupyter_server            1.17.0             pyhd8ed1ab_0    conda-forge
jupyterlab                3.4.2              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
jupyterlab_server         2.14.0             pyhd8ed1ab_0    conda-forge
kealib                    1.4.14               h87e4c3c_3    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.2           py310hbf28c38_1    conda-forge
krb5                      1.19.3               h3790be6_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
lerc                      3.0                  h9c3ff4c_0    conda-forge
libarchive                3.5.2                hccf745f_1    conda-forge
libblas                   3.9.0           14_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h166bdaf_7    conda-forge
libbrotlidec              1.0.9                h166bdaf_7    conda-forge
libbrotlienc              1.0.9                h166bdaf_7    conda-forge
libcap                    2.64                 ha37c62d_0    conda-forge
libcblas                  3.9.0           14_linux64_openblas    conda-forge
libclang                  13.0.1          default_hc23dcda_0    conda-forge
libcups                   2.3.3                hf5a7f15_1    conda-forge
libcurl                   7.83.1               h7bff187_0    conda-forge
libdap4                   3.20.6               hd7c4107_2    conda-forge
libdb                     6.2.32               h9c3ff4c_0    conda-forge
libdeflate                1.10                 h7f98852_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               h9b69904_4    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libflac                   1.3.4                h27087fc_0    conda-forge
libgcc-ng                 12.1.0              h8d9b700_16    conda-forge
libgdal                   3.4.3                h56144a5_0    conda-forge
libgfortran-ng            12.1.0              h69a702a_16    conda-forge
libgfortran5              12.1.0              hdcd56e2_16    conda-forge
libglib                   2.70.2               h174f98d_4    conda-forge
libgomp                   12.1.0              h8d9b700_16    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libkml                    1.3.0             h238a007_1014    conda-forge
liblapack                 3.9.0           14_linux64_openblas    conda-forge
liblapacke                3.9.0           14_linux64_openblas    conda-forge
libllvm11                 11.1.0               hf817b99_3    conda-forge
libllvm13                 13.0.1               hf817b99_2    conda-forge
libmamba                  0.24.0               hd8a31e3_1    conda-forge
libmambapy                0.24.0          py310hab0e683_1    conda-forge
libnetcdf                 4.8.1           nompi_h329d8a1_102    conda-forge
libnghttp2                1.47.0               h727a467_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libogg                    1.3.4                h7f98852_1    conda-forge
libopenblas               0.3.20          pthreads_h78a6416_0    conda-forge
libopus                   1.3.1                h7f98852_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libpq                     14.2                 hd57d9b9_0    conda-forge
librttopo                 1.1.0                hf69c175_9    conda-forge
libsndfile                1.0.31               h9c3ff4c_1    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libsolv                   0.7.22               h6239696_0    conda-forge
libspatialindex           1.9.3                h9c3ff4c_4    conda-forge
libspatialite             5.0.1               ha867d66_15    conda-forge
libssh2                   1.10.0               ha56f1ee_2    conda-forge
libstdcxx-ng              12.1.0              ha89aaad_16    conda-forge
libtiff                   4.4.0                h0fcbabc_0    conda-forge
libtool                   2.4.6             h9c3ff4c_1008    conda-forge
libudev1                  249                  h166bdaf_2    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libvorbis                 1.3.7                h9c3ff4c_0    conda-forge
libwebp                   1.2.2                h3452ae3_0    conda-forge
libwebp-base              1.2.2                h7f98852_1    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxkbcommon              1.0.3                he3ba5ed_0    conda-forge
libxml2                   2.9.12               h885dcf4_1    conda-forge
libxslt                   1.1.33               h0ef7038_3    conda-forge
libzip                    1.8.0                h4de3113_1    conda-forge
libzlib                   1.2.12               h166bdaf_0    conda-forge
llvmlite                  0.38.1          py310h58363a5_0    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lxml                      4.8.0           py310h5764c6d_2    conda-forge
lz4                       4.0.0           py310h5d5e884_2    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mamba                     0.24.0          py310hf87f941_1    conda-forge
mapclassify               2.4.3              pyhd8ed1ab_0    conda-forge
markdown                  3.3.7              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.1           py310h5764c6d_1    conda-forge
matplotlib                3.5.2           py310hff52083_0    conda-forge
matplotlib-base           3.5.2           py310h5701ce4_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
memory_profiler           0.60.0             pyhd8ed1ab_0    conda-forge
metis                     5.1.0             h58526e2_1006    conda-forge
mistune                   0.8.4           py310h6acc77f_1005    conda-forge
msgpack-python            1.0.4           py310hbf28c38_0    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mysql-common              8.0.29               haf5c9bc_1    conda-forge
mysql-libs                8.0.29               h28c427c_1    conda-forge
nbclassic                 0.3.7              pyhd8ed1ab_0    conda-forge
nbclient                  0.6.4              pyhd8ed1ab_1    conda-forge
nbconvert                 6.5.0              pyhd8ed1ab_0    conda-forge
nbconvert-core            6.5.0              pyhd8ed1ab_0    conda-forge
nbconvert-pandoc          6.5.0              pyhd8ed1ab_0    conda-forge
nbformat                  5.4.0              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
nest-asyncio              1.5.5              pyhd8ed1ab_0    conda-forge
netcdf4                   1.5.8           nompi_py310hd7ca5b8_101    conda-forge
networkx                  2.8.2              pyhd8ed1ab_0    conda-forge
nodeenv                   1.6.0              pyhd8ed1ab_0    conda-forge
nomkl                     1.0                  h5ca1d4c_0    conda-forge
notebook                  6.4.11             pyha770c72_0    conda-forge
notebook-shim             0.1.0              pyhd8ed1ab_0    conda-forge
nspr                      4.32                 h9c3ff4c_1    conda-forge
nss                       3.78                 h2350873_0    conda-forge
numba                     0.55.0          py310h00e6091_0  
numexpr                   2.8.0           py310hf05e7a9_102    conda-forge
numpy                     1.22.4          py310h4ef5377_0    conda-forge
openjpeg                  2.4.0                hb52868f_1    conda-forge
openpyxl                  3.0.9              pyhd8ed1ab_0    conda-forge
openssl                   1.1.1o               h166bdaf_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.4.2           py310h769672d_2    conda-forge
pandoc                    2.18                 ha770c72_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
panel                     0.13.1             pyhd8ed1ab_0    conda-forge
param                     1.12.1             pyh6c4a22f_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.2.0              pyhd8ed1ab_0    conda-forge
patsy                     0.5.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h9c3ff4c_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.1.1           py310he619898_1    conda-forge
pip                       22.1.2             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h36c2ea0_0    conda-forge
plac                      1.3.5              pyhd8ed1ab_0    conda-forge
platformdirs              2.5.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0           py310hff52083_3    conda-forge
ply                       3.11                       py_1    conda-forge
poppler                   22.04.0              h1434ded_1    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                14.2                 h2510834_0    conda-forge
powerplantmatching        0.5.3                    pypi_0    pypi
pre-commit                2.19.0          py310hff52083_0    conda-forge
progressbar2              4.0.0              pyhd8ed1ab_0    conda-forge
proj                      9.0.0                h93bde94_1    conda-forge
prometheus_client         0.14.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.29             pyha770c72_0    conda-forge
protobuf                  3.20.1                   pypi_0    pypi
psutil                    5.9.1           py310h5764c6d_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pulp                      2.6.0           py310hff52083_1    conda-forge
pulseaudio                14.0                 hbc9ff1d_7    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.3           py310h5764c6d_1010    conda-forge
pycountry                 20.7.3             pyh9f0ad1d_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyct                      0.4.6                      py_0    conda-forge
pyct-core                 0.4.6                      py_0    conda-forge
pygments                  2.12.0             pyhd8ed1ab_0    conda-forge
pyomo                     6.4.1           py310hd8f1fbe_0    conda-forge
pyopenssl                 22.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyproj                    3.3.1           py310h9bf108f_0    conda-forge
pypsa                     0.19.3             pyhd8ed1ab_0    conda-forge
pyqt                      5.15.4          py310hd8f1fbe_0    conda-forge
pyqt5-sip                 12.9.0          py310hd8f1fbe_0    conda-forge
pyrsistent                0.18.1          py310h5764c6d_1    conda-forge
pyshp                     2.3.0              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1           py310hff52083_5    conda-forge
pytables                  3.7.0           py310hf5df6ce_0    conda-forge
pytest                    7.1.2           py310hff52083_0    conda-forge
python                    3.10.4          h9a8a25e_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.15.3             pyhd8ed1ab_0    conda-forge
python-utils              3.3.3              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    2_cp310    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
pyviz_comms               2.2.0              pyhd8ed1ab_0    conda-forge
pyyaml                    6.0             py310h5764c6d_4    conda-forge
pyzmq                     23.1.0          py310h330234f_0    conda-forge
qt-main                   5.15.2               hdf1cb14_3    conda-forge
rasterio                  1.2.10          py310hdadb6e9_5    conda-forge
ratelimiter               1.2.0                   py_1002    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
reproc                    14.2.3               h7f98852_0    conda-forge
reproc-cpp                14.2.3               h9c3ff4c_0    conda-forge
requests                  2.27.1             pyhd8ed1ab_0    conda-forge
retry                     0.9.2                      py_0    conda-forge
reverse-geocode           1.4.1              pyhd8ed1ab_0    conda-forge
rioxarray                 0.11.1             pyhd8ed1ab_0    conda-forge
rtree                     1.0.0           py310hbdcdc62_1    conda-forge
ruamel_yaml               0.15.80         py310h5764c6d_1007    conda-forge
scikit-learn              1.1.1           py310hffb9edd_0    conda-forge
scipy                     1.8.1           py310h7612f91_0    conda-forge
seaborn                   0.11.2               hd8ed1ab_0    conda-forge
seaborn-base              0.11.2             pyhd8ed1ab_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                62.3.2          py310hff52083_0    conda-forge
shapely                   1.8.2           py310hb974679_1    conda-forge
sip                       6.5.1           py310h122e73d_2    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smart_open                6.0.0              pyhd8ed1ab_0    conda-forge
smmap                     3.0.5              pyh44b312d_0    conda-forge
snakemake-minimal         7.8.1              pyhdfd78af_0    bioconda
snappy                    1.1.9                hbd366e4_1    conda-forge
sniffio                   1.2.0           py310hff52083_3    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
sqlite                    3.38.5               h4ff8645_0    conda-forge
stack_data                0.2.0              pyhd8ed1ab_0    conda-forge
statsmodels               0.13.2          py310h96516ba_0    conda-forge
stopit                    1.1.2                      py_0    conda-forge
tabulate                  0.8.9              pyhd8ed1ab_0    conda-forge
tbb                       2021.5.0             h924138e_1    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
terminado                 0.15.0          py310hff52083_0    conda-forge
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
tiledb                    2.8.3                h1e4a385_1    conda-forge
tinycss2                  1.1.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.11.2             pyhd8ed1ab_0    conda-forge
toposort                  1.7                pyhd8ed1ab_0    conda-forge
tornado                   6.1             py310h5764c6d_3    conda-forge
tqdm                      4.64.0             pyhd8ed1ab_0    conda-forge
traitlets                 5.2.2.post1        pyhd8ed1ab_0    conda-forge
tsam                      2.1.0                    pypi_0    pypi
typing_extensions         4.2.0              pyha770c72_1    conda-forge
tzcode                    2022a                h166bdaf_0    conda-forge
tzdata                    2022a                h191b570_0    conda-forge
ukkonen                   1.0.1           py310hbf28c38_2    conda-forge
unicodedata2              14.0.0          py310h5764c6d_1    conda-forge
unidecode                 1.3.4                    pypi_0    pypi
urllib3                   1.26.9             pyhd8ed1ab_0    conda-forge
virtualenv                20.14.1         py310hff52083_0    conda-forge
vresutils                 0.3.1                    pypi_0    pypi
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.3.2              pyhd8ed1ab_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
wrapt                     1.14.1          py310h5764c6d_0    conda-forge
xarray                    2022.3.0           pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.3                h8ce2273_4    conda-forge
xlrd                      2.0.1              pyhd8ed1ab_3    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h7f98852_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxrender           0.9.10            h7f98852_1003    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xyzservices               2022.4.0           pyhd8ed1ab_0    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
yaml-cpp                  0.7.0                h27087fc_1    conda-forge
yte                       1.5.1           py310hff52083_0    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zict                      2.2.0              pyhd8ed1ab_0    conda-forge
zipp                      3.8.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.12               h166bdaf_0    conda-forge
zstd                      1.5.2                h8a70e8d_1    conda-forge
      ```
    </details>

### PR DESCRIPTION
**Pull request by davide-f**
Closes #241 

## Change proposed in this Pull Request

This PR aims at avoiding NaN values into the inflow.
Any NaN value would be converted into 0.0 instead before summing to the inflow.

## Description

When performing the shift of the inflow dataarray, some NaN values appears hence leading to NaN values into the sum.
This has been fixed.

## Motivation and Context

If NaN values remain, this may lead to issues in the usage of the tool

## How Has This Been Tested?

Using pypsa-africa

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [ ] I locally ran `pytest` inside the repository and no unexpected problems came up.
- [x] I have adjusted the docstrings in the code appropriately.
- [x] I have documented the effects of my code changes in the documentation `doc/`.
- [x] I have added newly introduced dependencies to `environment.yaml` file.
- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution

### CODE REVIEW

**Comment by fneum** [comment]
Thanks @davide-f for looking into this! Do you happen to know the reason for the NaN values? Would be good to treat the root cause rather than the symptoms.

File: atlite/hydro.py [hunk 1]
```diff
@@ -103,6 +103,13 @@ def shift_and_aggregate_runoff_for_plants(
...
         for b in ppl.upstream:
-            inflow_plant += runoff.sel(hid=b).shift(time=nhours.at[b])
+            # get upstream inflows
+            upstream_inflow_b = runoff.sel(hid=b).shift(time=nhours.at[b])
+            # replace null values with zeros
+            upstream_inflow_b = upstream_inflow_b.where(
+                upstream_inflow_b.isnull() == False, 0.0
+            )
+            # update inflow for the plant
+            inflow_plant += upstream_inflow_b
```
Comment (FabianHofmann) [inline atlite/hydro.py:113]:
```suggestion
            upstream_inflow_b = runoff.sel(hid=b).shift(time=nhours.at[b], fill_value=0)
```
wouldn't this already do it?

**Comment by davide-f** [comment]
> Thanks @davide-f for looking into this! Do you happen to know the reason for the NaN values? Would be good to treat the root cause rather than the symptoms.

The reason is the intrinsic shift operation: when a time series is shifted at the beginning (or end in case of negative shifts) NaN values are created because of missing data.
This link is pretty detailed: https://towardsdatascience.com/all-the-pandas-shift-you-should-know-for-data-analysis-791c1692b5e
There is no circulation in the shift, though it may be an interesting feature

File: atlite/hydro.py (hunk 1 above)
Comment (davide-f) [reply atlite/hydro.py]: Thank you! Much more elegant

**Comment by davide-f** [comment]
> > Thanks @davide-f for looking into this! Do you happen to know the reason for the NaN values? Would be good to treat the root cause rather than the symptoms.
> 
> The reason is the intrinsic shift operation: when a time series is shifted at the beginning (or end in case of negative shifts) NaN values are created because of missing data. This link is pretty detailed: https://towardsdatascience.com/all-the-pandas-shift-you-should-know-for-data-analysis-791c1692b5e There is no circulation in the shift, though it may be an interesting feature

Just checked that there is the function "roll" that performs the circular shift, shall we go for it?

**Comment by fneum** [comment]
> Just checked that there is the function "roll" that performs the circular shift, shall we go for it?

I'd suggest yes.

**Comment by davide-f** [comment]
> > Just checked that there is the function "roll" that performs the circular shift, shall we go for it?
> 
> I'd suggest yes.

Completely agree (already done), as not using it may lead to underestimate the hydro potential for a region.
Obviously, there may not be temporal correspondence (the flow of one region in 12/2012 will become available in 1/2012 when only 2012 is considered), but I'd say it is a super detail in this case.

This PR is ready

**Pull request merged**